File: 0398-Random Pick Index/0398-Random Pick Index.cpp
class Solution {
public:
    Solution(vector<int> nums) {
        this->nums = nums;
    }
    
    int pick(int target) {
        int index = -1;
        int count = 0;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (nums[i] == target)
            {
                if (rand() % (++count) == 0)
                {
                    index = i;
                }
            }
        }
        
        return index;
    }
    
private:
    vector<int> nums;
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int param_1 = obj.pick(target);
 */
 

File: 0301-0400/0398-Random Pick Index/0398-Random Pick Index.py
import random
class Solution:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def pick(self, target: int) -> int:
        index = -1
        count = 0
        for i, n in enumerate(self.nums):
            if n == target:
                if random.randrange(count + 1) == 0:
                    index = i
                count += 1
        return index


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)
